### PR TITLE
TRAFODION-1704: T2 and T4 driver cleanup

### PR DESCRIPTION
**Pull request by robertamarton**
Removed obsolete files from T2, redesigned result set code for T4 driver code
so same files can be removed.

Removed BaseRow.java, Row.java, and InsertRow.java from T2 and T4 code
Added ObjectArray.java to T4 code to store and manage an array of objects

TRAFODION-1642 Release 1.3.0 copyrights and license files are incorrect

Part 3 of cleaning up copyrights and license files

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -12,24 +12,25 @@ Hadoop.
...
 
-    - download site:
+    = download site:
       https://dist.apache.org/repos/dist/release/incubator/trafodion
 
-    - git site:
+    = git site:
       git@github.com:apache/incubator-trafodion  
 
   - To build and try out Apache Trafodion, please following our build
```
Comment (DaveBirdsall) [inline README.txt]: Grammar: "please following" should be "please follow"... can change on a subsequent commit if you like. Also, would be good to make the capitalization of the bullets consistent.

File: core/sql/common/multi-byte.h [hunk 2]
```diff
@@ -18,12 +18,6 @@
 // under the License.
 //
 // @@@ END COPYRIGHT @@@
-/*
```
Comment (DaveBirdsall) [inline core/sql/common/multi-byte.h]: Could you explain the provenance behind this code (and other similar files below) that allows us to remove these copyrights?

**Comment by DaveBirdsall** [comment]
Looks good to me. One question above.

File: core/sql/common/multi-byte.h (hunk 2 above)
Comment (robertamarton) [inline core/sql/common/multi-byte.h]: Hewlett Packard donated the original Trafodion code to Apache.  HP is a founding member of the Open Software Foundation and has all rights to code produced under this license.  Therefore these copyrights are covered by the same agreement that allows us to remove HP copyrights.  This information came from a conversion I had with the team that handled the software donation from HP to Apache.

**Comment by DaveBirdsall** [comment]
Thank you, Roberta. This looks ready to commit to me.

**Comment by robertamarton** [comment]
Changes have been made to rewrite portions of the t4 driver code.  With
this delivery  all existing tests have been run and passed. This check-in,
however, disables several jdbc methods mostly related to T4 row sets. We do
not have regression tests for these jdbc methods. Therefore, if someone
writes a test to use one of the disabled methods a “not supported” error is
returned.  The tests that fail use scroll sensitive and scroll insensitive
requests.



I was hoping that this delivery would become the next release candidate and
it would be submitted to Apache.  However,  a decision needs to be made if
we should release this with missing functionality.  We are working on
adding these missing methods.



Since this is the first Apache release, no one has built using the source
tar file so there should not be any regressions.  Also people can still go
to the git repository and build from the release 2.0 branch that has this
functionality.  When release 1.3 is merged into 2.0, we will make sure all
the methods work as desired.



Opinion:



+1 – go ahead and release with unsupported methods

0 – no opinion

-1 – add code for unsupported methods



    Roberta





*From:* DaveBirdsall [mailto:notifications@github.com]
*Sent:* Monday, December 14, 2015 11:13 AM
*To:* apache/incubator-trafodion <incubator-trafodion@noreply.github.com>
*Cc:* Roberta Marton <roberta.marton@esgyn.com>
*Subject:* Re: [incubator-trafodion] TRAFODION-1704: T2 and T4 driver
cleanup (#224)



Thank you, Roberta. This looks ready to commit to me.

—
Reply to this email directly or view it on GitHub
<https://github.com/apache/incubator-trafodion/pull/224#issuecomment-164529781>
.[image:
https://github.com/notifications/beacon/AMa1IBlwPGUiAkq-b8B2UE2dBNUpIBXRks5pPwwxgaJpZM4GzwUk.gif]

**Comment by robertamarton** [comment]
Let’s commit these changes so far.  If we decide to implement the
unsupported commands, then it will come as a new pull request.



    Roberta



*From:* Roberta Marton [mailto:roberta.marton@esgyn.com]
*Sent:* Monday, December 14, 2015 11:34 AM
*To:* 'apache/incubator-trafodion' <
reply@reply.github.com>;
'apache/incubator-trafodion' <incubator-trafodion@noreply.github.com>
*Subject:* RE: [incubator-trafodion] TRAFODION-1704: T2 and T4 driver
cleanup (#224)



Changes have been made to rewrite portions of the t4 driver code.  With
this delivery  all existing tests have been run and passed. This check-in,
however, disables several jdbc methods mostly related to T4 row sets. We do
not have regression tests for these jdbc methods. Therefore, if someone
writes a test to use one of the disabled methods a “not supported” error is
returned.  The tests that fail use scroll sensitive and scroll insensitive
requests.



I was hoping that this delivery would become the next release candidate and
it would be submitted to Apache.  However,  a decision needs to be made if
we should release this with missing functionality.  We are working on
adding these missing methods.



Since this is the first Apache release, no one has built using the source
tar file so there should not be any regressions.  Also people can still go
to the git repository and build from the release 2.0 branch that has this
functionality.  When release 1.3 is merged into 2.0, we will make sure all
the methods work as desired.



Opinion:



+1 – go ahead and release with unsupported methods

0 – no opinion

-1 – add code for unsupported methods



    Roberta





*From:* DaveBirdsall [mailto:notifications@github.com
<notifications@github.com>]
*Sent:* Monday, December 14, 2015 11:13 AM
*To:* apache/incubator-trafodion <incubator-trafodion@noreply.github.com>
*Cc:* Roberta Marton <roberta.marton@esgyn.com>
*Subject:* Re: [incubator-trafodion] TRAFODION-1704: T2 and T4 driver
cleanup (#224)



Thank you, Roberta. This looks ready to commit to me.

—
Reply to this email directly or view it on GitHub
<https://github.com/apache/incubator-trafodion/pull/224#issuecomment-164529781>
.[image:
https://github.com/notifications/beacon/AMa1IBlwPGUiAkq-b8B2UE2dBNUpIBXRks5pPwwxgaJpZM4GzwUk.gif]

**Pull request merged**